docs/source/working_with_aiida/caching.rst

```diff
@@ -65,88 +115,76 @@ This means that caching is enabled for ``TemplatereplacerCalculation`` and ``Str
+    In [5]: n2 = Str('test string')
 
-* The :meth:`~aiida.orm.nodes.Node.is_valid_cache` property determines whether a particular node can be used as a cache. This is used for example to disable caching from failed calculations.
-* Node classes have a ``_cachable`` attribute, which can be set to ``False`` to completely switch off caching for nodes of that class. This avoids performing queries for the hash altogether.
+    In [6]: n2.store(use_cache=True)
+    Out[6]: u'test string'
 
-There are two ways in which the hash match can go wrong: False negatives, where two nodes should have the same hash but do not, or false positives, where two different nodes have the same hash. It is important to understand that false negatives are **highly preferrable**, because they only increase the runtime of your calculations, as if caching was disabled. False positives however can break the logic of your calculations. Be mindful of this when modifying the caching behaviour of your calculation and data classes.
+    In [7]: print('UUID of n1:', n1.uuid)
+    UUID of n1: 956109e1-4382-4240-a711-2a4f3b522122
 
-.. _caching_error:
+    In [8]: print('n2 is cached from:', n2.get_cache_source())
+    n2 is cached from: 956109e1-4382-4240-a711-2a4f3b522122
 
-What to do when caching is used when it shouldn't
--------------------------------------------------
+When running a :class:`~aiida.engine.processes.CalcJob` through the :meth:`~aiida.engine.run` or :meth:`~aiida.engine.submit` functions, you can achieve the same effect using the :class:`~aiida.manage.caching.enable_caching` context manager:
 
-In general, the caching mechanism should trigger only when the output of a calculation will be exactly the same as if it is run again. However, there might be some edge cases where this is not true.
+.. code:: python
 
-For example, if the parser is in a different python module than the calculation, the version number used in the hash will not change when the parser is updated. While the "correct" solution to this problem is to increase the version number of a calculation when the behavior of its parser changes, there might still be cases (e.g. during development) when you manually want to stop a particular node from being cached.
+    from aiida.manage.caching import enable_caching
+    from aiida.engine import run
+    from aiida.orm import CalcJobNode
+    with enable_caching(node_class=CalcJobNode):
+       run(...)
 
-In such cases, you can follow these steps to disable caching:
+If you suspect a node is being reused in error (e.g. during development),
+it is also possible to manually *prevent* a specific node from being reused:
 
-1. If you suspect that a node has been cached in error, check that it has a ``_aiida_cached_from`` extra. If that's not the case, it is not a problem of caching.
-2. Get all nodes which match your node, and clear their hash:
+1. For one of the cloned nodes, check that :meth:`~aiida.orm.nodes.Node.get_cache_source` returns a UUID.
```

Comment:
For one of the _accidentally_ cloned nodes?

Or _unwantedly_ maybe? In any case, first time reading it was not clear that the it referenced a calculation that was cached from another that was not supposed to be used.

Reply:
Thanks, I've rephrased this.